packages/SwingSet/src/initializeSwingset.js

```diff
@@ -157,45 +159,57 @@ function resolveSpecFromConfig(dirname, specPath) {
  * path and make sure it has a `parameters` property if it's supposed to.
  *
  * @param {SwingSetConfigDescriptor | void} desc  The config descriptor to be normalized.
- * @param {string} dirname  The pathname of the directory in which the config file was found
+ * @param {string} referrer  The pathname of the file or directory in which the
+ * config file was found
  * @param {boolean} expectParameters `true` if the entries should have parameters (for
  *    example, `true` for `vats` but `false` for bundles).
  */
-function normalizeConfigDescriptor(desc, dirname, expectParameters) {
+async function normalizeConfigDescriptor(desc, referrer, expectParameters) {
+  const normalizeSpec = async (entry, key) => {
+    return resolveSpecFromConfig(referrer, entry[key]).then(spec => {
+      entry[key] = spec;
+    });
+  };
```

Comment:
I might have just used `await` in the for loop below, but the lint rule that forbids that is hinting that we can schedule work to be done in parallel instead. These are filesystem routines that probably don’t benefit from parallel interleaving, but what-the-heck. Who wants to be friends with the linter! 👈